src/components/Reports/GoalCalculator/SummaryReport/Steps/PresentingYourGoalStep/GoalApplicationButtonGroup.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import { Button, CircularProgress, Stack } from '@mui/material';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import { useUpdateAccountPreferencesMutation } from 'src/components/Settings/preferences/accordions/UpdateAccountPreferences.generated';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { Goal } from '../../MpdGoal/MpdGoalTable';
+import { useGoalLineItems } from '../../MpdGoal/useGoalLineItems';
+
+interface GoalApplicationButtonGroupProps {
+  goal: Goal;
+}
+
+export const GoalApplicationButtonGroup: React.FC<
+  GoalApplicationButtonGroupProps
+> = ({ goal }) => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { overallTotal } = useGoalLineItems(goal);
+  const [updateAccountPreferences, { loading }] =
+    useUpdateAccountPreferencesMutation();
+  const accountListId = useAccountListId() || '';
+  const [buttonsHidden, setButtonsHidden] = useState(false);
+
+  const onSave = async () => {
+    await updateAccountPreferences({
+      variables: {
+        input: {
+          id: accountListId,
+          attributes: {
+            id: accountListId,
+            settings: { monthlyGoal: overallTotal },
+          },
+        },
+      },
+      onCompleted: () => {
+        enqueueSnackbar(t('Successfully updated your monthly goal!'), {
```

Comment:
What do you think about telling the user what the goal was set to?

```suggestion
        enqueueSnackbar(t('Successfully updated your monthly goal to {{total}}!'), {
```